File: LeetCode/April Challenge/22nd Day/Day_22.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int,int> prevSum;
        int res=0;
        int currsum=0;
        
        for(int i=0;i<nums.size();i++)
        {
            currsum+=nums[i];
            if(currsum==k)
                res++;
            if(prevSum.find(currsum-k) != prevSum.end())
                res+=(prevSum[currsum-k]);
            prevSum[currsum]++;
        }
        return res;
    }
};

File: LeetCode/April Challenge/22nd Day/Day_22.py
from collections import defaultdict 
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        prevSum = defaultdict(lambda : 0)
        
        count, currSum = 0, 0
        
        for num in nums:
            currSum += num
            
            if currSum == k:
                count += 1
            
            if currSum - k in prevSum:
                count += prevSum[currSum - k]
                
            prevSum[currSum] += 1
        return count
        
